pkg/engine/provider/extract.go

```diff
@@ -0,0 +1,185 @@
+package provider
+
+import (
+	"context"
+	"io/fs"
+	"os"
+	"os/signal"
+	"path/filepath"
+	"sync"
+
+	"github.com/Checkmarx/kics/pkg/model"
+	"github.com/rs/zerolog/log"
+
+	"github.com/hashicorp/go-getter"
+)
+
+const (
+	channelLength = 2
+)
+
+// ExtractedPath is a struct that contains the paths, temporary paths to remove
+// and extraction map path of the sources
+// Path is the slice of paths to scan
+// ExtrectionMap is a map that correlates the temporary path to the givven path
+// RemoveTmp is the slice containing temporary paths to be removed
+type ExtractedPath struct {
+	Path          []string
+	ExtrectionMap map[string]model.ExtractedPathObject
+	RemoveTmp     []string
+}
+
+type getterStruct struct {
+	ctx         context.Context
+	cancel      context.CancelFunc
+	mode        getter.ClientMode
+	pwd         string
+	opts        []getter.ClientOption
+	destination string
+	source      string
+}
+
+// GetSources goes through the source slice, and determines the of source type (ex: zip, git, local).
+// It than extracts the files to be scanned. If the source given is not local, a temp dir
+// will be created where the files will be stored.
+func GetSources(source []string) (ExtractedPath, error) {
+	returnStr := ExtractedPath{
+		Path:          []string{},
+		ExtrectionMap: make(map[string]model.ExtractedPathObject),
+		RemoveTmp:     []string{},
+	}
+	for _, path := range source {
+		destination, err := os.MkdirTemp("", "kics-extract-*")
+		if err != nil {
+			return ExtractedPath{}, err
+		}
+
+		err = os.RemoveAll(destination)
+		if err != nil {
+			return ExtractedPath{}, err
+		}
+
+		mode := getter.ClientModeAny
+
+		pwd, err := os.Getwd()
+		if err != nil {
+			log.Fatal().Msgf("Error getting wd: %s", err)
+		}
+
+		opts := []getter.ClientOption{}
+
+		opts = append(opts, getter.WithInsecure())
+
+		ctx, cancel := context.WithCancel(context.Background())
+
+		st := getterStruct{
+			ctx:         ctx,
+			cancel:      cancel,
+			mode:        mode,
+			pwd:         pwd,
+			opts:        opts,
+			destination: destination,
+			source:      path,
+		}
+
+		dst, err := getPaths(&st)
+		if err != nil {
+			log.Error().Msgf("failed to find path %s: %s", path, err)
+			return ExtractedPath{}, err
+		}
+		returnStr.RemoveTmp = append(returnStr.RemoveTmp, dst)
+		tempDst, local := checkSymLink(dst, path)
+
+		returnStr.ExtrectionMap[dst] = model.ExtractedPathObject{
+			Path:      path,
+			LocalPath: local,
+		}
+
+		returnStr.Path = append(returnStr.Path, tempDst)
+	}
+
+	return returnStr, nil
```

Comment:
I think variables here could be a little more clear. Think `dst` or `st` does not represents anything in a clear way...